src/include/mpir_impl_tmp.h

```diff
@@ -3,8 +3,8 @@
  *     See COPYRIGHT in top-level directory
  */
 
-#ifndef MPIR_IMPL_H_INCLUDED
-#define MPIR_IMPL_H_INCLUDED
+#ifndef MPIR_IMPL_TMP_H_INCLUDED
```

Comment:
I feel we can separately review the generation of `mpir_impl.h` after the MPI/PMPI work is merged. Or is there some reason this needs to be done together?

Reply:
If we don't generate `mpir_impl.h`, then I have to manually declare the prototypes, as I did with `pt2pt` and during refactoring of `attr` and `errorhan`. That is the reason `mpir_impl.h` was created in the first place. We have to manually check which function needs the prototypes since many but not all in every chapter has slightly changed functions names (mostly with or without `_impl` suffix).  This is a very tedious work that will get deleted in the end. So I am hoping that we can avoid it.